### PR TITLE
Optimize for doc level function in group key generator

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Remove the error checking in generateKeyForDocId().
Optimize for single group-by column.
The overall performance gain for certain use case is up to 20%.

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupKeyGenerator.java [hunk 1]
```diff
@@ -170,16 +170,14 @@ public static void decodeRawGroupKey(long rawGroupKey, int[] cardinalities, int[
...
   @Override
   public int generateKeyForDocId(int docId) {
+    long rawKey = 0;
+
     for (int i = 0; i < _groupByColumns.length; i++) {
-      if (_isSingleValueGroupByColumn[i]) {
-        _singleValIterators[i].skipTo(docId);
-        _reusableGroupByValuesArray[i] = _singleValIterators[i].nextIntVal();
-      } else {
-        throw new RuntimeException("GroupBy on multi-valued columns not supported.");
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupKeyGenerator.java]: is this validation done during the initialization of keyGenerator?

**Comment by kishoreg** [comment]
LGTM. Make sure we have the pre condition while we are initializing the key generator

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupKeyGenerator.java (hunk 1 above)
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/DefaultGroupKeyGenerator.java]: Single group key and multi group keys share this class, and each of them calls different functions. This one is only called by single group key situation.

**Comment by kishoreg** [comment]
LGTM

**Comment by mayankshriv** [comment]
lgtm

**Pull request merged**